pkg/config/config_writer.go

```diff
@@ -664,8 +664,12 @@ func parseNameParameter(context *detailedSerializerContext, config Config) (conf
 	nameParam, found := config.Parameters[NameParameter]
 
 	if !found {
-		return nil, fmtDetailedConfigWriterError(context.serializerContext, "%s: `name` parameter missing",
-			config.Coordinate)
+		if config.Type.ID() == ClassicApiTypeId {
```

Comment:
for discussion: I don't think the config writer should even include this check, why would the writer care and throw errors if a config has a name or not

Reply:
agree lets remove it?

Reply:
removed in the commit that originally limited it to classic API types